File: src/main/java/nz/laspruca/tcplugin/eventlistners/BlockPlaceBreak.java
package nz.laspruca.tcplugin.eventlistners;

import nz.laspruca.tcplugin.logger.loggerevent.*;
import org.bukkit.*;
import org.bukkit.block.*;
import org.bukkit.entity.*;
import org.bukkit.event.*;
import org.bukkit.event.block.*;

import java.util.*;

import static org.qrl.tcplugin.TCPlugin.config;
import static org.qrl.tcplugin.TCPlugin.logger;

public class BlockPlaceBreak implements Listener {
	public static List<Material> breakBlocks = new ArrayList<>();
	public static List<Material> placeBlocks = new ArrayList<>();

	@EventHandler
	public static void onPlayerBreakBlock(BlockBreakEvent e) {
		HumanEntity player = e.getPlayer();
		Block block = e.getBlock();

		if (breakBlocks.contains(block.getType())) {
			new BreakBlockEvent(player.getName(), block).log();
		}
	}

	@EventHandler
	public static void onPlayerPlaceBlock(BlockPlaceEvent e) {
		HumanEntity player = e.getPlayer();
		Block block = e.getBlock();

		if (placeBlocks.contains(block.getType())) {
			new PlaceBlockEvent(player.getName(), block).log();
		}
	}


	public static void loadBlocks() {
		List<String> _break = config.getStringList("blocks-break");
		for (String a : _break) {
			try {
				breakBlocks.add(Material.valueOf(a.toUpperCase()));
			} catch (Exception e) {
				logger.warning("Invalid block in config " + a);
			}
		}


		breakBlocks.add(Material.CHEST);

		List<String> place = config.getStringList("place-break");
		for (String a : place) {
			try {
				placeBlocks.add(Material.valueOf(a.toUpperCase()));
			} catch (Exception e) {
				logger.warning("Invalid block in config " + a);
			}
		}
	}
}

File: src/main/kotlin/nz/laspruca/tcplugin/eventListeners/BlockPlaceBreak.kt
package nz.laspruca.tcplugin.eventListeners

import nz.laspruca.tcplugin.logger.BreakBlockEvent
import nz.laspruca.tcplugin.logger.PlaceBlockEvent
import org.bukkit.Material
import org.bukkit.entity.HumanEntity
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockBreakEvent
import org.bukkit.event.block.BlockPlaceEvent
import org.qrl.tcplugin.TCPlugin.Companion.config
import org.qrl.tcplugin.TCPlugin.Companion.logger
import java.lang.Exception
import java.util.ArrayList

private var breakBlocks: List<Material> = ArrayList()
private var placeBlocks: List<Material> = ArrayList()


fun loadBlocks() {
    breakBlocks = config.getStringList("blocks-break").mapNotNull {
        try {
            Material.valueOf(it.toUpperCase())
        } catch (_: java.lang.IllegalArgumentException) {
            logger.warning("No material for $it")
            null
        }
    }.toList()

    placeBlocks = config.getStringList("place-break").mapNotNull {
        try {
            Material.valueOf(it.toUpperCase())
        } catch (_: java.lang.IllegalArgumentException) {
            logger.warning("No material for $it")
            null
        }
    }.toList()
}

object BlockPlaceBreak : Listener {
    @EventHandler
    fun onPlayerBreakBlock(e: BlockBreakEvent) {
        val player: HumanEntity = e.player
        val block = e.block
        if (breakBlocks.contains(block.type)) {
            BreakBlockEvent(player.name, block).log()
        }
    }

    @EventHandler
    fun onPlayerPlaceBlock(e: BlockPlaceEvent) {
        val player: HumanEntity = e.player
        val block = e.block
        if (placeBlocks.contains(block.type)) {
            PlaceBlockEvent(player.name, block).log()
        }
    }
}